### PR TITLE
tests integ: VLAN as a slave of a bond

### PR DESCRIPTION
**Pull request by Nandan307**
Test that a VLAN interface has been sucessfully added as a
slave to a bond.

Signed-off-by: Nandan Kulkarni <nakulkar@redhat.com>

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: tests/integration/bond_test.py [hunk 1]
```diff
@@ -369,3 +381,10 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
         assertlib.assert_state(bond_state)
 
     assertlib.assert_absent('bond99')
+
+
+def test_create_vlan_as_slave_of_bond(bond99_vlan101):
+    bond_name = BOND99
+    slaves = [bond99_vlan101[Interface.KEY][0][Interface.NAME]]
```
Comment (EdDev) [inline tests/integration/bond_test.py]: `bond99_vlan101` represents two interfaces: a VLAN and its base, so how do you know what is defined at position `0`?

File: tests/integration/bond_test.py [hunk 2]
```diff
@@ -94,6 +96,16 @@ def bond99_with_slave(eth2_up):
         yield state
 
 
+@pytest.fixture
+def bond99_vlan101(eth1_up):
```
Comment (EdDev) [inline tests/integration/bond_test.py]: This fixture defines VLAN101 on `eth1`.
It has nothing to do with bond99.

File: tests/integration/bond_test.py [hunk 3]
```diff
@@ -369,3 +381,11 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
         assertlib.assert_state(bond_state)
 
     assertlib.assert_absent('bond99')
+
+
+def test_create_vlan_as_slave_of_bond(bond99_vlan101):
+    bond_name = BOND99
+    vlan_id = 101
+    slaves = ['{}.{}'.format(bond_name, vlan_id)]
+    with bond_interface(bond_name, slaves) as state:
```
Comment (EdDev) [inline tests/integration/bond_test.py]: This creates a bond99 over a slave that does not exist (bond99.101).

**Comment by EdDev** [comment]
@Nandan307 , we will keep this PR open for a few more days, but if no activity will occur we will close it. You will be able to work on it later on and re-open it when ready.

**Comment by EdDev** [comment]
The PR is closed due to no activity.
The task is tracked by https://nmstate.atlassian.net/browse/NMSTATE-204.

When ready to continue the work, you are welcome to re-open it or a new one.

**Pull request closed without merging**